app/controllers/posts_controller.rb

```diff
@@ -19,12 +19,23 @@ def new
 
   # GET /posts/1/edit
   def edit
+    @post = Post.find(params[:id])
+  end
+
+  def like_post
+    post = Post.find(params[:id])
+    if post.increment!(:likes_count)
+      flash[:notice] = "You liked the post!"
+    else
+      flash[:alert] = "There was something wrong liking the post"
+    end
   end
 
   # POST /posts
   # POST /posts.json
   def create
     @post = Post.new(post_params)
+    @post.likes_count = 0
```

Comment:
As a note this would be cleaner if it was moved to the migration so it's the default value.